### PR TITLE
Add set_referrer action to set referral codes for a wallet

### PR DESCRIPTION
**Pull request by Frixoe**

### CODE REVIEW

File: src/bin/set_referrer.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+This is an example of setting a refferal code for a wallet.
```
Comment (lmlmt) [inline src/bin/set_referrer.rs:2]: typo `refferal` to `referral` or just remove comment as the functionality is self-evident

File: src/bin/set_referrer.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+*/
+use ethers::signers::LocalWallet;
+
+use hyperliquid_rust_sdk::{BaseUrl, ExchangeClient};
+use log::info;
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+    // Add a testnet key which has deposited some funds on HL testnet
```
Comment (lmlmt) [inline src/bin/set_referrer.rs:12]: `// Key was randomly generated for testing and shouldn't be used with any real funds`

File: src/bin/set_referrer.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    let wallet: LocalWallet = "e908f86dbb4d55ac876378565aafeabc187f6690f046459397b17d9b9a19688e"
+        .parse()
+        .unwrap();
+
+    let ex = ExchangeClient::new(None, wallet, Some(BaseUrl::Testnet), None, None)
```
Comment (lmlmt) [inline src/bin/set_referrer.rs:17]: nit: `let exchange_client = ...`

File: src/bin/set_referrer.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        .await
+        .expect("Couldn't get the exchange client");
```
Comment (lmlmt) [inline src/bin/set_referrer.rs:19]: Just unwrap. This should fail fast if exchange client can't be initialized. You can use the other examples in `bin` as a template

File: src/bin/set_referrer.rs (hunk 1 above)
Comment (Frixoe) [reply src/bin/set_referrer.rs]: Done :saluting_face:

File: src/bin/set_referrer.rs (hunk 3 above)
Comment (Frixoe) [reply src/bin/set_referrer.rs]: Done :saluting_face:

File: src/bin/set_referrer.rs (hunk 4 above)
Comment (Frixoe) [reply src/bin/set_referrer.rs]: Done ser :saluting_face:

File: src/bin/set_referrer.rs (hunk 2 above)
Comment (Frixoe) [reply src/bin/set_referrer.rs]: :thinking: But the wallet needs to have trading enabled to be able to test setting referral code. Anyways, I've changed the comment :saluting_face:

**Pull request merged**